libtelemetry/src/main/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -548,4 +551,20 @@ void onEnterForeground() {
     startLocation();
     ProcessLifecycleOwner.get().getLifecycle().removeObserver(this);
   }
+
+  private boolean checkWakeUp() {
+    try {
+      ApplicationInfo appInformation = applicationContext.getPackageManager()
+        .getApplicationInfo(applicationContext.getPackageName(),
+        PackageManager.GET_META_DATA);
+      if (appInformation != null && appInformation.metaData != null) {
+        boolean adjustWakeUp = appInformation.metaData.getBoolean(KEY_META_DATA_WAKE_UP, false);
```

Comment:
I think the extraction of the `adjustWakeUp` flag could be moved to a method, not to duplicate the code in both classes. Does that make sense in this context?